Source/WebCore/dom/Document.cpp

```diff
@@ -10347,7 +10347,10 @@ void Document::updateIntersectionObservations(const Vector<WeakPtr<IntersectionO
         if (!observer)
             continue;
 
-        auto needNotify = observer->updateObservations(*this);
+        RefPtr thisFrame = frame();
+        ASSERT(thisFrame);
```

Comment:
You should null check and return here. Asserts which will immediately be followed by a crash aren't worth much.

Reply:
I change it so that we fetch the frame once before the loop, and return if the frame is `nullptr`